### PR TITLE
[draft] feat: resolve style field in package json

### PR DESCRIPTION
**Pull request by burhanuday**
Part of https://github.com/webpack/webpack/issues/14893

> supports style resolution in package.json and test it + test support loader usage

### CODE REVIEW

File: lib/StyleFieldPlugin.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				if (
+					!stylePath ||
+					typeof stylePath !== "string" ||
+					stylePath === "." ||
+					stylePath === "./"
+				) {
+					return callback();
+				}
+				if (
+					// this.options.forceRelative
```
Comment (burhanuday) [inline lib/StyleFieldPlugin.js:73]: todo: add support for `forceRelative`

File: lib/StyleFieldPlugin.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		resolver
+			.getHook(this.source)
+			.tapAsync("StyleFieldPlugin", (request, resolveContext, callback) => {
+				// if request.context.issuer is not a css file, don't resolve to style field
+				const isIssuerAStyleFile =
+					request.context &&
+					// @ts-ignore
+					request.context.issuer &&
+					// @ts-ignore
+					request.context.issuer.endsWith(".css");
```
Comment (burhanuday) [inline lib/StyleFieldPlugin.js:45]: todo: check for all known css extensions
todo2: allow consumer to configure extensions array

File: lib/StyleFieldPlugin.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	apply(resolver) {
+		const target = resolver.ensureHook(this.target);
+		resolver
+			.getHook(this.source)
+			.tapAsync("StyleFieldPlugin", (request, resolveContext, callback) => {
+				// if request.context.issuer is not a css file, don't resolve to style field
+				const isIssuerAStyleFile =
+					request.context &&
+					// @ts-ignore
+					request.context.issuer &&
```
Comment (burhanuday) [inline lib/StyleFieldPlugin.js:43]: todo: check if type can be improved

**Comment by alexander-akait** [comment]
@burhanuday It should not implemented here, we already support it in css-loader https://github.com/webpack-contrib/css-loader/blob/master/src/plugins/postcss-import-parser.js#L228, we just need to update defaults in webpack

**Comment by burhanuday** [comment]
Ah. Will update the defaults in webpack. Did not occur to me that we could just change the default options. Closing this PR

**Pull request closed without merging**